### PR TITLE
✨ Navigate away, if the user closes the Solution, on whose detail view he is currently on

### PR DESCRIPTION
**Pull request by robpal94**
---

**Changes:**

1. Navigates the user away from the detail view, if he closes the associated Solution

**Issues:**

PR: #1257 

---

See the [examples](#example) for inspiration.

## How can others test the changes?

> Describe how others can test your changes (you can remove this section for typo fixes etc.)

## PR-Checklist

Please check the boxes in this list after submitting your PR:

- [x] You can merge this PR **right now** (if not, please prefix the title with "WIP: ")
- [x] I've tested **all** changes included in this PR.
- [x] I've also reviewed this PR myself before submitting (e.g. for scrambled letters, typos, etc.).
- [x] I've rebased the `develop` branch with my branch before finishing this PR.
- [x] I've **summarized all changes** in a list above.
- [x] I've mentioned all **PRs, which relate to this one**.
- [x] I've prefixed my Pull Request title is according to [gitmoji guide](https://gitmoji.carloscuesta.me/).

## Example

<details>
<summary>
:warning: Before merging please click here to expand and follow the guide.
</summary>
<br>

Please use `:twisted_rightwards_arrows:` at the beginning of your merge commit title.

Example 1:

<pre>
<code>
:twisted_rightwards_arrows: :bug: Fix Wrong Text Decoration at ...
</code>
</pre>

To get your commit message, just copy the first part of this pull request.

Example 2:
<pre>
<code>
**Changes:**

- Fixes Wrong Text Decoration at ...
- Fixes some typos
- ...

**Issues:**

Closes #NumberOfFixedIssue

PR: #NumberOfThisPR
</code>
</pre>
</details>

## Emojis

<details>
<summary>
Expand for a list of most used Emojis.
</summary>
<br>

Please prefix your commit messages with an Emoji.

Ref: https://gitmoji.carloscuesta.me/

| Description              | Glyphe               | Emoji  |
|--------------------------|----------------------|--------|
| Bugfix                   | `:bug:`              | 🐛     |
| Configuration releated   | `:wrench:`           | 🔧     |
| Cosmetic                 | `:lipstick:`         | 💄     |
| Dependencies Downgrade   | `:arrow_down:`       | ⬇️     |
| Dependencies Upgrade     | `:arrow_up:`         | ⬆️     |
| Formatting               | `:art:`              | 🎨     |
| Improving Performance    | `:zap:`              | ⚡️     |
| Initial commit           | `:tada:`             | 🎉     |
| Linter                   | `:rotating_light:`   | 🚨     |
| Miscellaneous            | `:package:`          | 📦     |
| New Feature              | `:sparkles:`         | ✨     |
| Refactoring Code         | `:recycle:`          | ♻️     |
| Releasing / Version tags | `:bookmark:`         | 🔖     |
| Removing Stuff           | `:fire:`             | 🔥     |
| Tests                    | `:white_check_mark:` | ✅     |
| Work In Progress (WIP)   | `:construction:`     | 🚧     |

</details>

### CODE REVIEW

File: src/modules/solution-explorer/solution-explorer-list/solution-explorer-list.ts [hunk 1]
```diff
@@ -189,6 +210,18 @@ export class SolutionExplorerList {
     return filteredEntries;
   }
 
+  private _cleanupSolution(uri: string): void {
+   const indexOfSolutionToBeRemoved: number = this._getIndexOfSolution(uri);
+
+   const uriNotFound: boolean = indexOfSolutionToBeRemoved < 0;
+   if (uriNotFound) {
+      return;
+    }
```
Comment (SteffenKn) [inline src/modules/solution-explorer/solution-explorer-list/solution-explorer-list.ts]: Add some spacing

```suggestion
    }

```

File: src/modules/solution-explorer/solution-explorer-list/solution-explorer-list.ts [hunk 2]
```diff
@@ -4,6 +4,9 @@ import {IIdentity} from '@essential-projects/iam_contracts';
 import {IDiagram, ISolution} from '@process-engine/solutionexplorer.contracts';
 import {ISolutionExplorerService} from '@process-engine/solutionexplorer.service.contracts';
 
+import {EventAggregator, Subscription} from 'aurelia-event-aggregator';
+import {activationStrategy, NavigationInstruction, Redirect, Router} from 'aurelia-router';
```
Comment (SteffenKn) [inline src/modules/solution-explorer/solution-explorer-list/solution-explorer-list.ts]:
```suggestion
import {
    activationStrategy,
    NavigationInstruction,
    Redirect,
    Router,
} from 'aurelia-router';
```

File: src/modules/solution-explorer/solution-explorer-list/solution-explorer-list.ts (hunk 1 above)
Comment (SteffenKn) [reply src/modules/solution-explorer/solution-explorer-list/solution-explorer-list.ts]: I think you added the newline to the wrong spot

File: src/modules/solution-explorer/solution-explorer-list/solution-explorer-list.ts [hunk 3]
```diff
@@ -152,17 +161,29 @@ export class SolutionExplorerList {
...
    */
   public async closeSolution(uri: string): Promise<void> {
-    const indexOfSolutionToBeRemoved: number = this._getIndexOfSolution(uri);
 
-    const uriNotFound: boolean = indexOfSolutionToBeRemoved < 0;
-    if (uriNotFound) {
-      return;
-    }
+    /**
+     * If the user closes the Solution which contains the Diagram, which he still
```
Comment (alexanderkasten) [inline src/modules/solution-explorer/solution-explorer-list/solution-explorer-list.ts]:
```suggestion
     * If the user closes the Solution which contains the diagram, which he still
```

File: src/modules/solution-explorer/solution-explorer-list/solution-explorer-list.ts [hunk 4]
```diff
@@ -152,17 +161,29 @@ export class SolutionExplorerList {
...
+     * has opened, he gets navigated to the start page.
+     */
+    const currentOpenDiagram: string = this._router.currentInstruction.queryParams.solutionUri;
+    const diagramOfClosedSolutionOpen: boolean = uri.includes(currentOpenDiagram);
+
+    if (diagramOfClosedSolutionOpen) {
+      /**
+       * We only want to close the open Solution, if the User don't have
```
Comment (alexanderkasten) [inline src/modules/solution-explorer/solution-explorer-list/solution-explorer-list.ts]:
```suggestion
       * We only want to close the open Solution, if the user does not have
```

**Pull request merged**